llvm/lib/Target/AMDGPU/GCNSubtarget.h

```diff
@@ -443,6 +444,13 @@ class GCNSubtarget final : public AMDGPUGenSubtargetInfo,
     return isAmdHsaOS() ? TrapHandlerAbi::AMDHSA : TrapHandlerAbi::NONE;
   }
 
+  // True on hardware where 's_trap 2' is treated as a nop that must be
+  // simulated.
+  bool requiresSimulatedTrap() const {
+    AMDGPU::IsaVersion V = AMDGPU::getIsaVersion(getCPU());
+    return V.Major == 11 && V.Minor <= 3;
```

Comment:
Don't you want:
```suggestion
    return V.Major == 11 && V.Minor == 0 && V.Stepping <= 3;
```
?

Reply:
We shouldn't have to parse the target-cpu name in any query. Just define a proper feature for this?